scripts/test-runner.js

```diff
@@ -0,0 +1,208 @@
+                const outFD = fs.openSync('rc_stdout.log', 'w');
+                const errFD = fs.openSync('rc_stderr.log', 'w');
+                rcProcess = spawn('java', [
+                    `-Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY}`,
+                    '-cp',
+                    CLASSPATH,
+                    'com.hazelcast.remotecontroller.Main',
+                    '--use-simple-server'
+                ], {
+                    stdio: [
+                        'ignore',
+                        outFD,
+                        errFD
+                    ]
+                });
+                rcProcess.on('close', () => {
+                    fs.closeSync(outFD);
+                    fs.closeSync(errFD);
+                });
+            } catch (err) {
+                throw err;
+            }
+        }
+    }
+
+    console.log('Please wait for Hazelcast Remote Controller to start ...');
+
+    const retryCount = 10;
+
+    for (let i = 0; i < retryCount; i++) {
+        console.log('Trying to connect to 127.0.0.1:9701...');
+        const addressReachable = await isAddressReachable('127.0.0.1', 9701, 5000);
+        if (addressReachable) {
+            return;
+        }
+        await new Promise(r => setTimeout(r, 1000));
+    }
+    throw `Could not reach to 127.0.0.1:9701 after trying ${retryCount} times.`;
+}
+const shutdownProcesses = () => {
+    console.log('Stopping remote controller and test processes...');
+    shutdownRC();
+    if (ON_WINDOWS) {
+        spawnSync('taskkill', ['/pid', testProcess.pid, '/f', '/t']); // simple sigkill not enough on windows
+    } else {
+        if (testProcess && testProcess.exitCode === null) testProcess.kill('SIGKILL');
+    }
+};
+const shutdownRC = () => {
+    if (ON_WINDOWS) {
+        spawnSync('taskkill', ['/pid', rcProcess.pid, '/f', '/t']);
+    } else {
+        if (rcProcess && rcProcess.exitCode === null) rcProcess.kill('SIGKILL');
+    }
+}
+
+if (process.argv.length === 3) {
+    if (process.argv[2] === 'unit') {
+        testCommand = 'node node_modules/mocha/bin/mocha "test/unit/**/*.js"';
+        testType = 'unit';
+    } else if (process.argv[2] === 'integration') {
+        testCommand = 'node node_modules/mocha/bin/mocha "test/integration/**/*.js"';
+        testType = 'integration';
+    } else if (process.argv[2] === 'alltests') {
+        testCommand = 'node node_modules/mocha/bin/mocha "test/**/*.js"';
+        testType = 'alltests';
+    } else if (process.argv[2] === 'startrc') {
+        startRC(true).then(() => {
+            console.log('Hazelcast Remote Controller is started!');
+            process.exit(0);
+        }).catch(err => {
+            console.log('Could not start Hazelcast Remote Controller due to an error:');
+            throw err;
+        });
+    } else if (process.argv[2] === 'coverage') {
+        testCommand = 'node node_modules/nyc/bin/nyc node_modules/mocha/bin/_mocha "test/**/*.js" -- '
```

Comment:
Will these references to mocha and nyc will be a problem? I dont know may be due to change of its location in library

Reply:
They might change the location, but hopefully that doesn't happen in minor versions.